### DISCUSSION TITLE
uncovered Mutants

### DISCUSSION THREAD

**Discussion by mimmi20** [infection/infection#1715] [Q&A] [answered]
PHPUnit shows a test coverage of 100% in my tests, but infection shows 3 uncovered Mutants. Is there an option to show these mutants? 
I already tried `--debug` and `--log-verbosity=all`.

```shell
Run vendor/bin/phpunit -c phpunit.xml --verbose --coverage-clover=.build/coverage/clover.xml --coverage-text --coverage-xml=.build/coverage/coverage-xml --log-junit=.build/coverage/phpunit.junit.xml
PHPUnit 9.5.21 #StandWithUkraine

Runtime:       PHP 8.1.8 with Xdebug 3.1.4
Configuration: phpunit.xml

.............                                                     13 / 13 (100%)

Time: 00:00.132, Memory: 16.00 MB

OK (13 tests, 7 assertions)

Generating code coverage report in Clover XML format ... done [00:00.003]

Generating code coverage report in HTML format ... done [00:00.013]


Code Coverage Report:     
  2022-08-05 13:36:39     
                          
 Summary:                 
  Classes: 100.00% (3/3)  
  Methods: 100.00% (3/3)  
  Lines:   100.00% (32/32)

Mimmi20\Detector\ConfigProvider
  Methods: 100.00% ( 1/ 1)   Lines: 100.00% (  1/  1)
Mimmi20\Detector\DeviceDetectorFactory
  Methods: 100.00% ( 1/ 1)   Lines: 100.00% ( 29/ 29)
Mimmi20\Detector\Module
  Methods: 100.00% ( 1/ 1)   Lines: 100.00% (  2/  2)

Generating code coverage report in PHPUnit XML format ... done [00:00.016]

Run infection -s --min-covered-msi=84 --min-msi=84 --coverage=.build/coverage --logger-github --log-verbosity=all --debug

    ____      ____          __  _
   /  _/___  / __/__  _____/ /_(_)___  ____
   / // __ \/ /_/ _ \/ ___/ __/ / __ \/ __ \
 _/ // / / / __/  __/ /__/ /_/ / /_/ / / / /
/___/_/ /_/_/  \___/\___/\__/_/\____/_/ /_/

#StandWithUkraine

Infection - PHP Mutation Testing Framework version 0.26.13


Running initial test suite...

PHPUnit version: 9.5.21

.: killed, M: escaped, U: uncovered, E: fatal error, X: syntax error, T: timed out, S: skipped, I: ignored


Generate mutants...

Processing source code files: 0
UUU..M.M.M..M............
Escaped mutants:
================
...
```

**Comment by sidz** [comment, marked as answer]
Hi @mimmi20.

you can find all mutation in the log file (if log is enabled) or you can use _--show-mutations_ option to see all of them as part of console output.

The full set of available options you can find here: https://infection.github.io/guide/command-line-options.html#show-mutations-or-s

**Reply by mimmi20** [reply]
Hi @sidz.

In my sample this option is set, but the uncovered mutants are not shown. As it is running in a Github pipeline, I can't into the log files

**Reply by sidz** [reply]
As I can see there is no `--show-mutations` option in your example. Are you sure that you've tried to use it?

see https://github.com/infection/infection/blob/master/src/Event/Subscriber/MutationTestingConsoleLoggerSubscriber.php#L98-L104

**Reply by mimmi20** [reply]
There is `-s` (the short form of --show-mutations) in the sample.

see [https://infection.github.io/guide/command-line-options.html#show-mutations-or-s](https://infection.github.io/guide/command-line-options.html#show-mutations-or-s)

**Reply by maks-rafalko** [reply]
that's very strange. Can you reproduce this locally?

**Reply by mimmi20** [reply]
The uncodered mutants are in the log file, but not in console output.

I m working with Windows, and I dont get XDebug running on my System. Therefore I normally dont run infection locally.

**Reply by mimmi20** [reply]
I need to add `-vv` to get the uncovered Mutants.

**Reply by maks-rafalko** [reply]
so do you say that escaped mutants are only shown on console with `-vv`? If so, this is a bug and I would ask you to create an issue with filled in template (OS, PHP/Infection version, etc)
